10_rest/runnerdave/cmd/puppy_server/main_test.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	puppy "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy/store"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	puppy1 = func() puppy.Puppy {
+		return puppy.Puppy{ID: 1, Breed: "Chihuahua", Colour: "Brown", Value: 12.30}
+	}
+	puppy2 = func() puppy.Puppy {
+		return puppy.Puppy{ID: 2, Breed: "Cacri", Colour: "Undefined", Value: 1.30}
+	}
+	puppy3 = func() puppy.Puppy {
+		return puppy.Puppy{ID: 12, Breed: "Imaginary", Colour: "Undefined", Value: -1.30}
+	}
+)
+
+func TestMain(m *testing.M) {
+	args = []string{}
+	os.Exit(m.Run())
+}
+
+func TestLoad(t *testing.T) {
+	s := store.NewSyncStore()
+
+	puppies := []puppy.Puppy{puppy1(), puppy2()}
+	err := load(s, puppies)
+	if err != nil {
+		panic(err)
+	}
+
+	readable := strconv.Quote(`{"breed":"Chihuahua","color":"Brown","id":1,"value":12.3} 
+	{"breed":"Cacri","color":"Undefined","id":2,"value":1.3} `)
+	expected := strings.ReplaceAll(readable, "\\n\\t", "")
+	actual := strconv.Quote(s.String())
+	t.Logf("expected:%s", expected)
+	t.Logf("actual:%s", actual)
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()")
+	}
+
```

Comment:
Unnecessary blank line.
No need for blank lines at beginning or end of block.
Use blank lines between type, method and function definitions and within a block for logical separation.
below as well.

Reply:
fixed